### DISCUSSION TITLE
Why is "Predicted PV" so high?

### DISCUSSION THREAD

**Discussion by mudge** [springfall2008/batpred#1202] [Q&A] [answered]
I have integrated predbat with Solcast via the [Predbat direct method](https://springfall2008.github.io/batpred/install/#solcast-install) for a 7.2 kWp PV system and a 5 kWh inverter (following the advice for hybrid inverters in the documentation and setting both my AC and DC capacity to 7.2 kW) but my predictions seem to show an unexpectedly high predicted PV value beyond what I could reasonably generate:

<img width="322" alt="Screenshot 2024-06-10 at 09 01 55" src="https://github.com/springfall2008/batpred/assets/287/e8487af9-c761-42af-af07-56f686132d37">

Looking at the HA states, I see the following entities:

| Entity | State |
| --- | --- |
| `predbat.base10_pv_energy` | 27.043 |
| `predbat.best10_pv_energy` | 27.043 |
| `predbat.best_pv_energy` | 69.22 |
| `predbat.pv_energy` | 69.22 |

The forecast solar production from HA's default Energy dashboard has a `sensor.energy_production_today` value of 26.285 kWh which seems more realistic.

Is predbat saying it is predicting 69 kWh of production today or am I misreading what predbat's `pv_energy` means?

**Comment by gcoan** [comment, marked as answer]
pv_energy is the '50%' forecast from Solcast.

I think the reason for this is that predbat is predicting the PV energy for today AND to the end of the forecast plan.  All of the predbat future forward sensors work like this, they are not just 'today' forecasts.

If I look at my own predbat.pv_energy, its 122kWh, way more than a day's generation.
If I look at the attributes it is forecasting from now (17:20 on 15/6) right through to 17:15 on 17/6:
![image](https://github.com/springfall2008/batpred/assets/142018870/66137f95-b8ba-4e0b-aa79-2353cca0b748)
![image](https://github.com/springfall2008/batpred/assets/142018870/25d0fefa-faa0-437b-b3bb-06946fdeaf29)

The forecast is also at 5 minute granularity which is I guess what it gets from Solcast

**Reply by mudge** [reply]
I think you're right and I was missing that the predicted PV energy is for the _entire prediction duration_ (which is roughly 48 hours).

**Reply by gcoan** [reply]
I've added a clarification about the forecase_hours to the documentation.

Current draft here: https://github.com/gcoan/batpred/blob/main/docs/output-data.md
